framework/quota/src/org/apache/cloudstack/quota/dao/QuotaTariffDaoImpl.java

```diff
@@ -0,0 +1,137 @@
+        searchUsageType.done();
+
+        listAllIncludedUsageType = createSearchBuilder();
+        listAllIncludedUsageType.and("onorbefore", listAllIncludedUsageType.entity().getEffectiveOn(), SearchCriteria.Op.LTEQ);
+        listAllIncludedUsageType.and("quotatype", listAllIncludedUsageType.entity().getUsageType(), SearchCriteria.Op.EQ);
+        listAllIncludedUsageType.done();
+    }
+
+    @Override
+    public QuotaTariffVO findTariffPlanByUsageType(final int quotaType, final Date effectiveDate) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        List<QuotaTariffVO> result = new ArrayList<>();
+        try (TransactionLegacy txn = TransactionLegacy.open(TransactionLegacy.USAGE_DB)) {
+            final Filter filter = new Filter(QuotaTariffVO.class, "effectiveOn", false, 0L, 1L);
+            final SearchCriteria<QuotaTariffVO> sc = listAllIncludedUsageType.create();
+            sc.setParameters("onorbefore", effectiveDate);
+            sc.setParameters("quotatype", quotaType);
+            result = search(sc, filter);
+        } catch (Exception e) {
+            throw new CloudRuntimeException("Unable to find tariff plan by usage type");
+        } finally {
+            TransactionLegacy.open(opendb).close();
+        }
+        if (result.size() > 0) {
+            if (s_logger.isDebugEnabled()){
+                s_logger.debug("QuotaTariffDaoImpl::findTariffPlanByUsageType: " + effectiveDate + "quota type " + quotaType  + " val=" + result.get(0).getCurrencyValue());
+            }
+            return result.get(0);
+        } else {
+            if (s_logger.isDebugEnabled()){
+                s_logger.info("QuotaTariffDaoImpl::findTariffPlanByUsageType: Missing quota type " + quotaType);
+            }
+            return null;
+        }
+    }
+
+    @Override
+    public List<QuotaTariffVO> listAllTariffPlans(final Date effectiveDate) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        List<QuotaTariffVO> tariffs = new ArrayList<QuotaTariffVO>();
+        try (TransactionLegacy txn = TransactionLegacy.open(TransactionLegacy.USAGE_DB)) {
+            final Filter filter = new Filter(QuotaTariffVO.class, "effectiveOn", false, 0L, 1L);
+            final SearchCriteria<QuotaTariffVO> sc = listAllIncludedUsageType.create();
+            sc.setParameters("onorbefore", effectiveDate);
+            for (Integer quotaType : QuotaTypes.listQuotaTypes().keySet()) {
+                sc.setParameters("quotatype", quotaType);
+                List<QuotaTariffVO> result = search(sc, filter);
+                if (result.size() > 0) {
+                    tariffs.add(result.get(0));
+                    s_logger.info("listAllTariffPlans onorbefore" + effectiveDate + "quota type " + result.get(0).getDescription() + " , effective Date=" + result.get(0).getEffectiveOn() + " val=" + result.get(0).getCurrencyValue());
```

Comment:
This log message lacks the context to be actionable by a sysadmin as an ``INFO`` message.  Either add the context necessary for actionability by a sysadmin and change to ``DEBUG``.